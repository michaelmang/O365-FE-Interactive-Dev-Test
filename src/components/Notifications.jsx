import PropTypes from 'prop-types';
import React from 'react';

import styles from './Notifications.module.scss';

const Notifications = ({ errors }) => (
  <div className={styles.container}>
    {errors.map(renderError)}
  </div>
);

Notifications.propTypes = {
  errors: PropTypes.arrayOf(PropTypes.shape({
    message: PropTypes.string,
  })).isRequired,
};

const renderError = ({ message }, idx) => (
  <div key={idx} className="notification is-danger">
    {message}
  </div>
);

export default Notifications;
