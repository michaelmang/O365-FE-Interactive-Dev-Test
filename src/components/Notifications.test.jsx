import { shallow } from 'enzyme';
import React from 'react';

import Notifications from './Notifications.jsx';
import renderSnapshot from '../testing/renderSnapshot.js';

jest.mock('./Notifications.module.scss', () => ({
  container: 'styles.container',
}));

describe('Notifications', () => {
  it('matches the saved snapshot', () => {
    let { root: actual } = extractComponents(renderComponent());

    expect(renderSnapshot(actual)).toMatchSnapshot();
  });

  const extractComponents = (root) => ({
    root,
  });

  const renderComponent = (props) => (
    shallow(
      <Notifications
        errors={[
          { id: 'some-error', message: 'derp' },
          { id: 'another-error', message: 'chirp' },
          { id: 'third-error', message: 'hero' },
        ]}
      />
    )
  );
});
