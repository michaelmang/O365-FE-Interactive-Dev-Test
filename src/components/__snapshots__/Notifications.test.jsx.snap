// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications matches the saved snapshot 1`] = `
<div
  className="styles.container"
>
  <div
    className="notification is-danger"
  >
    derp
  </div>
  <div
    className="notification is-danger"
  >
    chirp
  </div>
  <div
    className="notification is-danger"
  >
    hero
  </div>
</div>
`;
